src/Api.php

```diff
@@ -405,12 +406,26 @@ public function rest($pattern, $model = null, $methods = null)
                     $model = $this->call($model, $args);
                 }
 
-                // limit fields
-                $model->onlyFields($this->getAllowedFields($model, 'read'));
+                $this->loadModelByValue($model, $id);
+
+                //calculate only once
+                $allowed_fields = $this->getAllowedFields($model, 'read');
+                $data = [];
+                // get all field-elements
+                foreach ($model->elements as $field => $f) {
```

Comment:
No this is wrong now. Should use `$model->getFields()` and that way get rid of `instanceof \atk4\data\Field` condition and as result this change can be made simpler.

Reply:
Also tests fail.